python/cog/server/http.py

```diff
@@ -326,27 +334,20 @@ async def _predict(
         if respond_async:
             return JSONResponse(jsonable_encoder(initial_response), status_code=202)
 
-        # by now, output Path and File are already converted to str
-        # so when we validate the schema, those urls get cast back to Path and File
-        # in the previous implementation those would then get encoded as strings
-        # however the changes to Path and File break this and return the filename instead
-        try:
-            prediction = await async_result
-            # we're only doing this to catch validation errors
-            response = PredictionResponse(**prediction.dict())
-            del response
-        except ValidationError as e:
-            _log_invalid_output(e)
-            raise HTTPException(status_code=500, detail=str(e)) from e
-
-        # dict_resp = response.dict()
-        # output = await runner.client_manager.upload_files(
-        #     dict_resp["output"], upload_url
-        # )
-        # dict_resp["output"] = output
-        # encoded_response = jsonable_encoder(dict_resp)
-
-        # return *prediction* and not *response* to preserve urls
+        # # by now, output Path and File are already converted to str
+        # # so when we validate the schema, those urls get cast back to Path and File
+        # # in the previous implementation those would then get encoded as strings
+        # # however the changes to Path and File break this and return the filename instead
+        # try:
+        #     prediction = await async_result
+        #     # we're only doing this to catch validation errors
+        #     response = PredictionResponse(**prediction.dict())
+        #     del response
+        # except ValidationError as e:
+        #     _log_invalid_output(e)
+        #     raise HTTPException(status_code=500, detail=str(e)) from e
```

Comment:
What are the implications of commenting this out?

Reply:
outputs are not validated, no error is raised if you return a string instead of an int. this turned out to be a bottleneck with sufficient throughput in profiling, and also led to a bug mentioned in the comments where Path is converted to str but then validation converts it back to Path and this round-trip no longer works the way it used to